### PR TITLE
Samples 4 maps with defaults take2

### PR DESCRIPTION
**Pull request by AlexCue987**
add samples for `withDefault` and `getValue`

### CODE REVIEW

File: libraries/stdlib/samples/test/samples/collections/maps.kt [hunk 1]
```diff
@@ -206,6 +206,34 @@ class Maps {
...
+        @Sample
+        fun getValueWithDefault() {
+            val mapWithDefault = mapOf(1 to "One", 2 to "Two").withDefault { key ->
+                when {
+                    key == 0 -> "Zero"
+                    key > 0 -> "Positive"
+                    else -> "Negative"
+                }
+            }
+            val actual = (-1..3).associateBy( { it }, { mapWithDefault.getValue(it) } )
```
Comment (ilya-g) [inline libraries/stdlib/samples/test/samples/collections/maps.kt:225]: More idiomatic to use `associateWith` in this case, e.g. `(-1..3).associateWith { mapWithDefault.getValue(it) }`

File: libraries/stdlib/src/kotlin/collections/MapWithDefault.kt [hunk 2]
```diff
@@ -45,6 +48,9 @@ public fun <K, V> Map<K, V>.withDefault(defaultValue: (key: K) -> V): Map<K, V>
  * and a value is obtained with [Map.getValue] function, for example when properties are delegated to the map.
  *
  * When this map already has an implicit default value provided with a former call to [withDefault], it is being replaced by this call.
+ *
+ * @sample samples.collections.Maps.Usage.getValueWithDefault
+ * @sample samples.collections.Maps.Usage.getValueWithReplacedDefault
```
Comment (ilya-g) [inline libraries/stdlib/src/kotlin/collections/MapWithDefault.kt:53]: Makes sense to add another sample specific for MutableMaps with default, showing that the changes made to the returned map with a default are propagated back to the original mutable map.

**Review by ilya-g — CHANGES_REQUESTED** [review]

File: libraries/stdlib/src/kotlin/collections/MapWithDefault.kt (hunk 2 above)
Comment (AlexCue987) [reply libraries/stdlib/src/kotlin/collections/MapWithDefault.kt:53]: added `changesToMutableMapWithDefaultPropagateToUnderlyingMap`

File: libraries/stdlib/samples/test/samples/collections/maps.kt (hunk 1 above)
Comment (AlexCue987) [reply libraries/stdlib/samples/test/samples/collections/maps.kt]: done

**Review by ilya-g — APPROVED** [review]

**Comment by fzhinkin** [comment]
Merged: https://github.com/JetBrains/kotlin/commit/5e61072f69c18ca0de56e9e45b217d21c9013378

**Pull request closed without merging**